pkg/controllers/ingress.go

```diff
@@ -47,25 +48,23 @@ type IngressController struct {
 	edgelbManager manager.EdgeLBManager
 	// logger is the logger that the controller will use.
 	logger log.FieldLogger
+	// secretsReflector is the instance used to manage lifecycle of DC/OS secrets
+	secretsReflector secretsreflector.SecretsReflector
 }
 
 // NewIngressController creates a new instance of the EdgeLB ingress controller.
-func NewIngressController(kubeClient kubernetes.Interface, er record.EventRecorder, ingressInformer extsv1beta1informers.IngressInformer, serviceInformer corev1informers.ServiceInformer, kubeCache dklbcache.KubernetesResourceCache, edgelbManager manager.EdgeLBManager) *IngressController {
+func NewIngressController(kubeClient kubernetes.Interface, er record.EventRecorder, ingressInformer extsv1beta1informers.IngressInformer, serviceInformer corev1informers.ServiceInformer, kubeCache dklbcache.KubernetesResourceCache, edgelbManager manager.EdgeLBManager, secretsReflector secretsreflector.SecretsReflector) *IngressController {
```

Comment:
Again, this would be good to use a struct for input parameter, but not a blocker.

Reply:
Will add it to a PR I'm working on to refactor the code to add more unit tests.